hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -319,6 +320,68 @@ public static List<String> getAllPartitionPaths(HoodieEngineContext engineContex
     }
   }
 
+  /**
+   * Gets all partition paths in a table, based on a relative partition path prefix.
+   *
+   * @param engineContext {@link HoodieEngineContext} instance for file listing.
+   * @param hadoopConf    Hadoop configuration.
+   * @param basePath      Base path of the table.
+   * @param prefix        Relative partition path prefix for listing.
+   * @param parallelism   Listing parallelism to use if the engine context supports it.
+   * @return A list of  all partition paths with the prefix.
+   * @throws IOException upon error.
+   */
+  public static List<String> getPartitionPathsWithPrefix(HoodieEngineContext engineContext,
```

Comment:
we have FSUtils.getAllPartitionPaths(engineContext, basePath, false, cfg.assumeDatePartitioning) already. Don't think calling prefix based api is the right thing to do here.